src/test/java/com/artipie/RpPermissionsTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import java.io.File;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.AllOf;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.MatcherOf;
+
+/**
+ * Test for {@link RpPermissions}.
+ * @since 0.2
+ * @checkstyle LeftCurlyCheck (500 lines)
+ */
+@Disabled
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class RpPermissionsTest {
+
+    /**
+     * Test configuration file location.
+     */
+    private static final String CONF_YAML = "src/test/resources/repo_permissions.yml";
+
+    @Test
+    void johnCanDownloadDeployAndDelete() {
+        final String uname = "John";
+        MatcherAssert.assertThat(
+            new RpPermissions(new File(RpPermissionsTest.CONF_YAML)),
+            new AllOf<>(
+                new ListOf<>(
+                    new MatcherOf<>(perm -> { return perm.allowed(uname, "delete"); }),
```

Comment:
@g4s8 it cannot, unfortunately, because of the ambiguous reference: 
```
Error:(56, 21) java: reference to MatcherOf is ambiguous
  both constructor <T>MatcherOf(org.cactoos.Proc<T>) in org.llorllale.cactoos.matchers.MatcherOf and constructor <T>MatcherOf(org.cactoos.Func<T,java.lang.Boolean>) in org.llorllale.cactoos.matchers.MatcherOf match
```